src/CommunityToolkit.Maui/Views/Popup/Popup.shared.cs

```diff
@@ -1,415 +1,134 @@
-	/// that varies from the actual <see cref="Size"/> of the <see cref="Popup" />
-	/// </remarks>
-	public Size Size
-	{
-		get => (Size)GetValue(SizeProperty);
-		set => SetValue(SizeProperty, value);
-	}
+	public virtual Task Close(CancellationToken token = default) => GetPopupPage().Close(new PopupResult(false), token);
 
-	/// <summary>
-	/// Gets or sets a value indicating whether the popup can be dismissed by tapping outside the Popup.
-	/// </summary>
-	/// <remarks>
-	/// When true and the user taps outside the popup, it will dismiss.
-	/// On Android - when false the hardware back button is disabled.
-	/// </remarks>
-	public bool CanBeDismissedByTappingOutsideOfPopup
+	internal void NotifyPopupIsOpened()
 	{
-		get => (bool)GetValue(CanBeDismissedByTappingOutsideOfPopupProperty);
-		set => SetValue(CanBeDismissedByTappingOutsideOfPopupProperty, value);
+		Opened?.Invoke(this, EventArgs.Empty);
 	}
 
-	/// <summary>
-	/// Gets or sets the <see cref="Style"/> of the Popup.
-	/// </summary>
-	public Style Style
+	internal void NotifyPopupIsClosed()
 	{
-		get => (Style)GetValue(StyleProperty);
-		set => SetValue(StyleProperty, value);
+		Closed?.Invoke(this, EventArgs.Empty);
 	}
 
-	/// <summary>
-	/// Gets or sets the <see cref="View"/> anchor.
-	/// </summary>
-	/// <remarks>
-	/// The Anchor is where the Popup will render closest to. When an Anchor is configured,
-	/// the popup will appear centered over that control or as close as possible.
-	/// </remarks>
-	public View? Anchor { get; set; }
-
-	/// <summary>
-	/// Property that represents the Window that's showing the Popup.
-	/// </summary>
-	public Window Window
+	private protected PopupPage GetPopupPage()
 	{
-		get;
-		set
-		{
-			field = value;
+		var parent = Parent;
 
-			if (Content is IWindowController controller)
-			{
-				controller.Window = value;
-			}
-		}
-	} = new();
-
-	/// <summary>
-	/// Property that represents Resources of Popup.
-	/// </summary>
-	public ResourceDictionary Resources
-	{
-		get => resources;
-		set
+		while (parent is not null)
 		{
-			ArgumentNullException.ThrowIfNull(value);
-
-			if (resources == value)
+			if (parent.Parent is PopupPage popuppage)
 			{
-				return;
+				return popuppage;
 			}
 
-			OnPropertyChanging();
-			((IResourceDictionary)resources).ValuesChanged -= OnResourcesChanged;
-
-			resources = value;
-
-			OnResourcesChanged(value);
-			((IResourceDictionary)resources).ValuesChanged += OnResourcesChanged;
-
-			OnPropertyChanged();
-		}
-	}
-
-	/// <summary>
-	/// Property that represents Style Class of Popup.
-	/// </summary>
-	[System.ComponentModel.TypeConverter(typeof(ListStringTypeConverter))]
-	public IList<string> StyleClass
-	{
-		get => mergedStyle.StyleClass;
-		set => mergedStyle.StyleClass = value;
-	}
-
-	/// <summary>
-	/// Gets or sets the result that will return when the user taps outside the Popup.
-	/// </summary>
-	protected object? ResultWhenUserTapsOutsideOfPopup { get; set; }
```

Comment:
Did we really want to remove the ability to set a result for this scenario?

Reply:
If we reinstate this then we could potentially remove the need for the `WasDismissedByTappingOutsideOfPopup` property in `PopupResult`

Reply:
It's not needed because when the user taps outside of Popup, the result is always `null`.

Reply:
The only way to set the result of a Popup is to call `Popup.Close()`. When the user taps outside of the popup, the developer does not call `Popup.Close()` and therefore never provides us the result.

Reply:
To be honest, if I could wave a magic wand, I would remove `Popup<T>` completely because it just turns Popup into a Messaging Service which is entirely unnecessary. There already exists a million ways for developers to propagate data between the View and ViewModel.

Reply:
The developer doesn't call close but our code does? Could we somehow retrieve the default value from PopupOptions and return that value. Writing this it sounds a little involved and the only way I can think to do it is via an attached property

Reply:
> The developer doesn't call close but our code does 

Correct. We add a `GestureRecognizer` to the `PopupPage` that fires the `await Close(new PopupResult(true));` when the user taps outside of the popup.

https://github.com/CommunityToolkit/Maui/blob/e009ed47b82ded5b2ec27ed5413a7b6f11597498/src/CommunityToolkit.Maui/Views/Popup/PopupPage.shared.cs#L34-L63

Reply:
Oh this could be an easy solve then... we could sub class TapGestureRecognizer to take the default return value and return the value when it's actioned?

If it's something we want to retain I'm happy to make the changes.

I'm just working through all the current breaks and trying to see what future bug reports we can avoid

Reply:
Sorry - are we still talking about `ResultWhenUserTapsOutsideOfPopup` here, or did we cross threads?

It's a good breaking change to remove `ResultWhenUserTapsOutsideOfPopup`.

Reply:
I was still talking about that property. OK I can update the docs to reflect this change